src/app/mesh/qgsmaptooleditmeshframe.cpp

```diff
@@ -203,6 +211,56 @@ void QgsMeshEditForceByLineAction::updateSettings()
   settings.setEnumValue( QStringLiteral( "UI/Mesh/ForceByLineToleranceUnit" ), mUnitSelecionWidget->unit() );
 }
 
+//
+// QgsMeshEditDigitizingAction
+//
+
+QgsMeshEditDigitizingAction::QgsMeshEditDigitizingAction( QObject *parent )
+  : QWidgetAction( parent )
+{
+  QGridLayout *gLayout = new QGridLayout();
+  gLayout->setContentsMargins( 3, 2, 3, 2 );
+
+  QgsSettings settings;
+
+  QLabel *labelZValueType = new QLabel( tr( "Z value from" ) );
+  mComboZValueType = new QComboBox();
+  mComboZValueType->addItem( tr( "Z Widget" ), ZWidget );
+  mComboZValueType->addItem( tr( "Terrain" ), Terrain );
+  mComboZValueType->addItem( tr( "In Mesh Use Mesh Otherwise Terrain" ), InMeshUseMeshTerrainOtherwise );
+  mComboZValueType->addItem( tr( "In Mesh Use Mesh Otherwise Z Widget" ), InMeshUseMeshZWidgetOtherwise );
+
+  int interpolateFromValue = settings.enumValue( QStringLiteral( "UI/Mesh/ZValueFrom" ), InMeshUseMeshZWidgetOtherwise );
```

Comment:
settings are usually lowerCamelCase
```suggestion
  int interpolateFromValue = settings.enumValue( QStringLiteral( "UI/Mesh/zValueFrom" ), InMeshUseMeshZWidgetOtherwise );
```